clients/spark/build.sbt

```diff
@@ -103,12 +103,36 @@ assembly / assemblyShadeRules := Seq(
   rename("reactor.util.**").inAll
 )
 
-s3Upload / mappings := Seq(
-  (assembly / assemblyOutputPath).value ->
-    s"${name.value}/${version.value}/${(assembly / assemblyJarName).value}"
-)
-s3Upload / s3Host := "treeverse-clients-us-east.s3.amazonaws.com"
-s3Upload / s3Progress := true
+// Safe upload: put-object with If-None-Match to prevent overwriting an existing key.
+// Fails the build on any error and surfaces the AWS CLI error message.
+lazy val s3Upload = taskKey[Unit]("Upload JAR to S3 without override existing")
+val publishBucket = settingKey[String]("Target S3 bucket for publishing the JAR")
+
+// Overridable via -Dpublish.bucket
+publishBucket := sys.props.get("publish.bucket").getOrElse("treeverse-clients-us-east")
```

Comment:
Prefer to have the code handle defaults and empty the same. it will simplify the changes in the makefile and make the code there easy to pass just empty string.

```scala
sys.props.get("publish.bucket").filter(_.nonEmpty).getOrElse("treeverse-clients-us-east")
```